compiler/compiler.go

```diff
@@ -489,9 +507,33 @@ func (c *compileCtx) compileMap(name string, s *openapi.Schema) (protobuf.Type,
 		}
 	case !s.Type.Empty():
 		var err error
-		typ, err = c.getType(s.Type.First())
-		if err != nil {
-			return nil, errors.Wrapf(err, `failed to get type %s`, s.Type)
+		if s.Type.First() == "array" && s.Items != nil {
+			// we need to construct a new statically typed wrapper message that contains a repeated list of items
+			// referenced by the spec
+			mapValueName := getMapValueName(name)
+			baseFieldName := camelCase(strings.TrimPrefix(s.Items.Ref, "#/definitions"))
+			m := protobuf.NewMessage(mapValueName)
+			f := protobuf.NewField(protobuf.NewMessage(baseFieldName), rawName, 1)
+			f.SetRepeated(true)
+			if v := s.Description; len(v) > 0 {
+				f.SetComment(v)
+			}
+			m.AddField(f)
+			m.SetComment("automatically generated wrapper for a list of " + baseFieldName + " items")
+			typ = m
+
+			// finally, make sure that this type is registered, if need be.
+			// hack alert - I couldn't figure out how to stop the wrappers from being specified more than once.
+			// listValueWrappers was my fix :(
```

Comment:
Could we perhaps make this map a field on the `compileCtx`? Perhaps we could create a generic catcher of duplicate types and encapsulate it within `addTypeToParent`?